### PR TITLE
Lessen code duplication

### PR DESCRIPTION
**Pull request by Nhowka**
Just to show a possible modification that can be done if we drop the custom handlers that are adding a lot of complexity.

### CODE REVIEW

**Comment by Zaid-Ajaj** [comment]
This made my day :heart:

**Comment by Nhowka** [comment]
I probably can put the custom handlers back without adding complexity back to the server implementations, but we should?

**Comment by Zaid-Ajaj** [comment]
I believe they should be removed, I know, I have been talking a lot about removing stuff, I just want to tidy up the library and get rid of half solutions to problems we shouldn't be concerned with. If someone really wants to use custom handlers and, then why bother with Fable.Remoting in the first place, it is very counter-productive, especially for users, don't you agree?

**Comment by Nhowka** [comment]
I'll add them again just to be sure that it didn't break any test, but after that, we could make it go away for good.

**Comment by Nhowka** [comment]
The build is green. Honestly, I'm proud of the result but not sure if that would make it much easier to create new server implementations. Suave and Giraffe are too much alike. Maybe is a good time to do the Nancy one just to be sure if it will work.

**Comment by Zaid-Ajaj** [comment]
> Maybe is a good time to do the Nancy one just to be sure if it will work

Actually, it will be a lot simpler than you think, because all we need to do is write an aspnet core middleware and someone will be able to use Remoting anywhere, whether it is Nancy, Giraffe or even Freya etc.
```fs
app.UseNancy()
      .UseGiraffe(giraffeApp)
      .UseFreya(freyaApp)  
      .UseRemoting(musicStore, options) // something like this
```
I still haven't given this much thought, but it shouldn't be a problem, even if we were to implement it from scratch with a little bit of code duplication so don't worry about this being *not very extensible*

**Comment by Zaid-Ajaj** [comment]
See how giraffe implements [middleware](https://github.com/giraffe-fsharp/Giraffe/blob/master/src/Giraffe/Middleware.fs) for reference. 

We could be extremely lazy and use giraffe under the hood 
```fs
type IApplicationBuilder with 
  member this.UseRemoting<T>(T serverImpl) = 
    let webApp = remoting serverImpl {()} 
    this.UseGiraffe webApp 
```
Nah, just kidding :smile:

**Comment by Nhowka** [comment]
@Zaid-Ajaj Did what can become a middleware, but didn't test it yet...

File: Fable.Remoting.Middleware/FableMiddlewareAdapter.fs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    member this.WithLogger(l) = { this with Logger = Some l }
+    member this.WithErrorHandler(eh) = {this with ErrorHandler = Some eh}
+
+  type RemotingOptions = BuilderOptions<HttpContext>
+
+  type FableRemotingMiddleware(next          : RequestDelegate,
+                               implementation: obj,
+                               options       : RemotingOptions) =
+
+      do if isNull next then raise (ArgumentNullException("next"))
```
Comment (irium) [inline Fable.Remoting.Middleware/FableMiddlewareAdapter.fs]: Just a little improvement: `raise (ArgumentNullException("next"))` could be written as `nullArg "next"`

Comment (Nhowka) [reply Fable.Remoting.Middleware/FableMiddlewareAdapter.fs]: Done, thanks!

**Comment by Zaid-Ajaj** [comment]
@Nhowka Looks like it fits nicely with what is already done in `Server` :smile: You can merge and I can do the testing, since this is asp.net core, we can easily do the [integration testing using the dotnet client](https://zaid-ajaj.github.io/Fable.Remoting/src/dotnet-http-client.html)

**Pull request merged**